src/ray/gcs/redis_context.cc

```diff
@@ -595,6 +598,14 @@ Status RedisContext::Connect(const std::string &address,
 
   RAY_CHECK(!context_);
   RAY_CHECK(!redis_async_context_);
+
+  // Remember function arguments for reconnection
+  address_ = address;
```

Comment:
On every `Connect`, data member are copied from themselves, DNS record is re-resolved, is that expected and ideal?

I'm thinking if we should only initialize these members for once.
I understand DNS resolution's result could change, but considering the fact that it's heavily cached by your OS and DNS resolution devices, and you're not flushing records, the chance of change between invocations are ignorable.